From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 22 Mar 2019 20:34:14 +0100
Subject: [PATCH] {MAYBE UPS} drivers/net/wireless/intel/iwlwifi: fix type in
 compaison

arm-gcc complains:

    In file included from arch/arm/include/asm/div64.h:127,
                     from include/linux/kernel.h:18,
                     from include/linux/skbuff.h:17,
                     from include/linux/if_ether.h:23,
                     from include/linux/etherdevice.h:25,
                     from drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c:62:
    drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c: In function
    'iwl_mvm_debug_range_resp':
    include/asm-generic/div64.h:222:28: error: comparison of distinct
    pointer types lacks a cast [-Werror]
      (void)(((typeof((n)) *)0) == ((uint64_t *)0)); \
                                ^~
    drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c:465:2: note:
    in expansion of macro 'do_div'
      do_div(rtt_avg, 6666);
      ^~~~~~
---
 drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c b/drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c
index 2039b871d219..af9ca37e99bc 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/ftm-initiator.c
@@ -460,7 +460,7 @@ static int iwl_mvm_ftm_range_resp_valid(struct iwl_mvm *mvm, u8 request_id,
 static void iwl_mvm_debug_range_resp(struct iwl_mvm *mvm, u8 index,
 				     struct cfg80211_pmsr_result *res)
 {
-	s64 rtt_avg = res->ftm.rtt_avg * 100;
+	u64 rtt_avg = res->ftm.rtt_avg * 100;
 
 	do_div(rtt_avg, 6666);
 
-- 
